### PR TITLE
fix: Fix `increaseAllowance` bugs

### LINKED ISSUE #23884: [Bug]: IncreaseAllowance - Site Suggested value is empty and cannot proceed
### Describe the bug

Whenever I use the new increase allowance functionalitly for ERC20 tokens, when using the Site suggested value nothing happens and the Next button is disabled.

### Expected behavior

We should see the Site suggested value

### Screenshots/Recordings

https://github.com/MetaMask/metamask-extension/assets/54408225/57ead976-47e6-4a0e-8a2f-a8375599fb95

### Steps to reproduce

1. Go to the test dapp
2. Deploy an ERC20 token
3. Click Increase Allowance
4. Click Site Suggested
5. See empty value in the input field

### Error messages or log output

_No response_

### Version

maybe in 11.14.0 - cherry-pick

### Build type

None

### Browser

Chrome

### Operating system

Linux

### Hardware wallet

_No response_

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by pedronfigueiredo**
## **Description**

Fixes two bugs: one related to the site suggested value and another to the function name on the view details panel. See linked issues for repro instructions.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/23897?quickstart=1)

## **Related issues**

Fixes: #23884 and #23885

## **Manual testing steps**

1. Go to test dApp
2. Deploy an erc20 contract
3. Click increase token allowance

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/pages/confirmations/components/approve-content-card/approve-content-card.js [hunk 1]
```diff
@@ -154,9 +157,7 @@ export default function ApproveContentCard({
                 color={TextColor.textAlternative}
                 as="h6"
               >
-                {isSetApproveForAll
-                  ? t('functionSetApprovalForAll')
-                  : t('functionApprove')}
+                {`Function: ${functionName}`}
```
Comment (cryptotavares) [inline ui/pages/confirmations/components/approve-content-card/approve-content-card.js:160]: Do we need translactions here? Maybe not for the fucntion name, but for the word `Function`?

**Comment by cryptotavares** [comment]
Should we add an e2e test for the site suggested value?

**Review by cryptotavares — APPROVED** [review]

File: ui/pages/confirmations/components/approve-content-card/approve-content-card.js [hunk 2]
```diff
@@ -45,6 +46,11 @@ export default function ApproveContentCard({
 }) {
   const t = useContext(I18nContext);
 
+  const tokenData = parseStandardTokenTransactionData(data);
+  const functionName = tokenData?.name;
+  const capitalizedFnName =
+    functionName?.charAt(0).toUpperCase() + functionName?.slice(1);
```
Comment (digiwand) [inline ui/pages/confirmations/components/approve-content-card/approve-content-card.js:52]: double-checking, `!isSetApproveForAll` → `t('functionApprove')` seemed to be the catch-all case. It's okay to map the method name directly and there is no need for this catch-all anymore correct?

**Review by digiwand — APPROVED** [review]

File: ui/pages/confirmations/components/approve-content-card/approve-content-card.js (hunk 2 above)
Comment (pedronfigueiredo) [reply ui/pages/confirmations/components/approve-content-card/approve-content-card.js:52]: Yes, this card is only shown for certain transaction types, and we're getting the function name from the ABIs, which means all cases should be covered.

**Pull request merged**